lib/ui/painting/paint.cc

```diff
@@ -71,14 +71,11 @@ DlColor ReadColor(const tonic::DartByteData& byte_data) {
   // Invert alpha so 0 initialized buffer has default value;
   float alpha = 1.f - float_data[kColorAlphaIndex];
   uint32_t colorspace = uint_data[kColorSpaceIndex];
-  (void)colorspace;
-  uint32_t encoded_color =
-      static_cast<uint8_t>(std::round(alpha * 255.f)) << 24 |  //
-      static_cast<uint8_t>(std::round(red * 255.f)) << 16 |    //
-      static_cast<uint8_t>(std::round(green * 255.f)) << 8 |   //
-      static_cast<uint8_t>(std::round(blue * 255.f)) << 0;
-  // TODO(gaaclarke): Pass down color info to DlColor.
-  return DlColor(encoded_color);
+
+  DlColor dl_color(alpha, red, green, blue,
+                   static_cast<DlColorSpace>(colorspace));
+
+  return dl_color.withColorSpace(DlColorSpace::kExtendedSRGB);
```

Comment:
I think you also need the gradient conversions too, right?

Reply:
You're jumping ahead, that happens in https://github.com/flutter/engine/pull/54748.

Reply:
The conversion happens in a different place for gradients.  It was easier to do it in dart for them.  We can reevaluate after we get our integration tests landed.  It's ideal to do the conversion in c++.

Reply:
I think that you need to move it into the engine eventually so that we can handle android, which may have a displayP3 color space and not an extended srgb space.